classes/security/UserGroupDAO.inc.php

```diff
@@ -778,47 +779,82 @@ function deleteSettingsByLocale($locale) {
 	 * @return string SQL search snippet
 	 */
 	function _getSearchSql($searchType, $search, $searchMatch, &$params) {
+		$hasUserSetting = "EXISTS(
+			SELECT 0
+			FROM user_settings
+			WHERE user_id = u.user_id
+				AND setting_name = '%s'
+				AND LOWER(setting_value) LIKE LOWER(?)
+		)";
 		$searchTypeMap = [
-			IDENTITY_SETTING_GIVENNAME => 'usgs.setting_value',
-			IDENTITY_SETTING_FAMILYNAME => 'usfs.setting_value',
-			USER_FIELD_USERNAME => 'u.username',
-			USER_FIELD_EMAIL => 'u.email',
-			USER_FIELD_AFFILIATION => 'us.setting_value',
+			IDENTITY_SETTING_GIVENNAME => sprintf($hasUserSetting, IDENTITY_SETTING_GIVENNAME),
+			IDENTITY_SETTING_FAMILYNAME => sprintf($hasUserSetting, IDENTITY_SETTING_FAMILYNAME),
+			USER_FIELD_USERNAME => 'LOWER(u.username) LIKE LOWER(?)',
+			USER_FIELD_EMAIL => 'LOWER(u.email) LIKE LOWER(?)',
+			USER_FIELD_AFFILIATION => sprintf($hasUserSetting, USER_FIELD_AFFILIATION)
 		];
 
 		$searchSql = '';
-
+		$search = trim($search);
 		if (!empty($search)) {
-
 			if (!isset($searchTypeMap[$searchType])) {
-				$str = $this->concat('COALESCE(usgs.setting_value,\'\')', 'COALESCE(usfs.setting_value,\'\')', 'u.email', 'COALESCE(us.setting_value,\'\')');
-				$concatFields = ' ( LOWER(' . $str . ') LIKE ? OR LOWER(cves.setting_value) LIKE ? ) ';
-
-				$search = strtolower($search);
-
-				$words = preg_split('{\s+}', $search);
-				$searchFieldMap = array();
-
-				foreach ($words as $word) {
-					$searchFieldMap[] = $concatFields;
-					$term = '%' . $word . '%';
-					array_push($params, $term, $term);
+				$terms = array_map(function ($term) {
+					return "%$term%";
+				}, PKPString::regexp_split('/\s+/', $search));
+				$filters = [];
+
+				switch (get_class(Capsule::connection())) {
+					case MySqlConnection::class:
+						$concatSettingValue = "GROUP_CONCAT(setting_value SEPARATOR '')";
+						break;
+					case PostgresConnection::class:
+						$concatSettingValue = "STRING_AGG(setting_value, '')";
+						break;
+					default:
+						throw new DomainException('Unrecognized database');
 				}
```

Comment:
This is the second time I've added such customization, perhaps there's a library providing such extensions out there (I doubt)... If not, perhaps we should polyfill the Laravel Database or centralize these customizations somewhere.